CppImport/src/macro/StaticStuff.cpp

```diff
@@ -0,0 +1,358 @@
+		qDebug() << "not removed" << node->typeName() << "in" << node->parent()->typeName();
+}
+
+void StaticStuff::addNodeToDeclaration(Model::Node* node, OOModel::Declaration* declaration)
+{
+	if (auto ooExpression = DCast<OOModel::Expression>(node))
+	{
+		if (auto context = DCast<OOModel::Method>(declaration))
+			context->items()->append(new OOModel::ExpressionStatement(ooExpression));
+		else
+			qDebug() << "can not insert expression" << ooExpression->typeName() << "into a" << declaration->typeName();
+	}
+	else if (auto ooStatement = DCast<OOModel::Statement>(node))
+	{
+		if (auto context = DCast<OOModel::Method>(declaration))
+			context->items()->append(ooStatement);
+		else
+			Q_ASSERT(false);
+	}
+	else if (auto ooDeclaration = DCast<OOModel::Declaration>(node))
+	{
+		if (auto ooClass = DCast<OOModel::Class>(node))
+		{
+			if (auto context = DCast<OOModel::Project>(declaration))
+				context->classes()->append(ooClass);
+			else if (auto context = DCast<OOModel::Module>(declaration))
+				context->classes()->append(ooClass);
+			else if (auto context = DCast<OOModel::Class>(declaration))
+				context->classes()->append(ooClass);
+			else
+				Q_ASSERT(false);
+		}
+		else if (auto ooField = DCast<OOModel::Field>(node))
+		{
+			if (auto context = DCast<OOModel::Project>(declaration))
+				context->fields()->append(ooField);
+			else if (auto context = DCast<OOModel::Module>(declaration))
+				context->fields()->append(ooField);
+			else if (auto context = DCast<OOModel::Class>(declaration))
+				context->fields()->append(ooField);
+			else
+				Q_ASSERT(false);
+		}
+		else if (auto ooVarDecl = DCast<OOModel::VariableDeclaration>(node))
+		{
+			if (auto context = DCast<OOModel::Method>(declaration))
+				context->items()->append(new OOModel::ExpressionStatement(
+													 new OOModel::VariableDeclarationExpression(ooVarDecl)));
+			else
+				Q_ASSERT(false);
+		}
+		else if (auto ooMethod = DCast<OOModel::Method>(node))
+		{
+			if (auto context = DCast<OOModel::Project>(declaration))
+				context->methods()->append(ooMethod);
+			else if (auto context = DCast<OOModel::Module>(declaration))
+				context->methods()->append(ooMethod);
+			else if (auto context = DCast<OOModel::Class>(declaration))
+				context->methods()->append(ooMethod);
+			else
+				Q_ASSERT(false);
+		}
+		else
+		{
+			if (auto context = DCast<OOModel::Project>(declaration))
+				context->subDeclarations()->append(ooDeclaration);
+			else if (auto context = DCast<OOModel::Module>(declaration))
+				context->subDeclarations()->append(ooDeclaration);
+			else if (auto context = DCast<OOModel::Class>(declaration))
+				context->subDeclarations()->append(ooDeclaration);
+			else if (auto context = DCast<OOModel::Method>(declaration))
+				context->subDeclarations()->append(ooDeclaration);
+			else
+				Q_ASSERT(false);
+		}
+	}
+	else
+		Q_ASSERT(false && "not implemented");
+}
+
+QVector<Model::Node*> StaticStuff::topLevelNodes(QVector<Model::Node*> input)
+{
```

Comment:
Use stl algorithms to simplify your work and make it quicker:
    
    {
       auto end = std::remove_if(input.begin(), input.end(), [=](auto e){
           for(auto other: input)
             if (other != e && other->isAncestorOf(e)) return true;
           return false;
         }
    input.erase(end, input.end());
    return input;